### PR TITLE
Add deleteRecursivelyOnExit()

### PR DESCRIPTION
**Pull request by lbergelson**
this adds a shutdown hook to delete a directory and its contents and works even on directories that didn't exist at call time

### CODE REVIEW

**Comment by lbergelson** [comment]
@davidadamsphd This should solve your problem.

File: .gitignore [hunk 1]
```diff
@@ -17,3 +17,6 @@ gradlew.bat
 #Please don't commit me
 client_secret.json
 servicekey.json
+
+#Test Generated File
+I_SHOULD_HAVE_BEEN_DELETED
```
Comment (davidadamsphd) [inline .gitignore]: Zuh?

Comment (davidadamsphd) [inline .gitignore]: Do we want to keep that in there?

**Comment by davidadamsphd** [comment]
LGTM, modulo my one question. Submit whenever you're ready.

File: .gitignore (hunk 1 above)
Comment (lbergelson) [inline .gitignore]: I'm fine with strange things in .gitignore.

**Pull request merged**